tests/test_adjoint_jacobian.py

```diff
@@ -53,6 +54,17 @@
 )
 
 
+def get_random_state(n):
+    np.random.seed(GLOBAL_SEED)
+    return np.random.rand(n) + 1j * np.random.rand(n)
```

Comment:
```suggestion
    np.random.seed(GLOBAL_SEED)
    random_state = np.random.rand(n) + 1j * np.random.rand(n)
    return random_state / np.linalg.norm(random_state)
```

It seems like it would be better to normalize here rather than on each invocation of the function